### PR TITLE
Luis add warnings component to dashboard

### PR DESCRIPTION
**Pull request by luisarevalo21**
# Description
Screenshot 2024-01-23 at 8 49 55 PM

The main functions I created was: 
A way to track warnings issued by an Admin or Owner account. 
8 Icons are displayed and can be selected assigning a warning
After 2 warnings assigend a modal will appear to provide options.
Issuing a warning will change the color to yellow (Future PR will send an email currently only an alert is issued)
Issuing a blue square will change the color to red (Future PR will create and assign a blue square)
Selecting a colored icon will delete it, a modal will appear to confirm the deletion. 

## Related PRS (if any):
This frontend PR is related to the https://github.com/OneCommunityGlobal/HGNRest/pull/718 backend PR.
…

## Main changes explained:
Created a warnings component inside of components
All notable files related to the component are inside of warnings. 
`Warnings.jsx` is the main file and maps over the user's warnings received from the backend.
The warnings are also filtered by blue yellow red then by date. 

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Login as an admin account.
4. Once the dashboard loads, you'll see a toggle button underneath every user
5. click the toggle button to view the users' warnings
6. Click an icon to assign a warning to the user' corresponding to the description. 
7. Once they receive more than 2 a modal will appear. 
8. 4 options are given issue warning, log warning only, cancel and issue blue square
9. log warning only will color the icon blue like normal
10. issue warning will color the icon yellow an alert will appear saying an email was sent (Future PR)
11. Issuing a blue square will color red and an alert will appear saying blue square was sent (Future PR)
12. Selecting a colored allows you to delete it.
13. You may toggle the button to remove the warnings component and will appear one you click it again. 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/ff425741-6b7f-4472-a5cb-cec7f3e2d56e

## Note:
This is the basis for the warnings component. 
I'll create other PR to add more warning descriptions, send emails, and issue blue square in the future. I just wanted to lay out a basis first then add to it. 
the following warnings appear and are unrelated to the PR 
Screenshot 2024-01-23 at 8 20 35 PM

### CODE REVIEW

**Review by KevHnh — CHANGES_REQUESTED** [review]
Appears that a specific backend must be used to review this PR. I get an error message when clicking on the "Tracking" button. I will review again when I get an update. 

image

**Comment by luisarevalo21** [comment]
> Appears that a specific backend must be used to review this PR. I get an error message when clicking on the "Tracking" button. I will review again when I get an update.
> 
> image

@KevHnh 
Yes you must check into the backend the PR is https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/718

Please check in and follow the steps.

**Review by Renanluizssx — COMMENTED** [review]
Hi @luisarevalo21, i tested your application and i didn't get to see the team members tasks, look my review please:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/6846a641-4592-4fa8-bcc7-675d27482f3c

**Comment by Cavein254** [comment]
@luisarevalo21 Tested the pr and failed to receive the results as explained. The functionality fails with status code `404`


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13707896/1a1c4fd0-5406-4d2b-83af-a2b48a21c53a

**Review by SRamen1999 — CHANGES_REQUESTED** [review]
After testing this PR I came across some problems. When using the backend PR https://github.com/OneCommunityGlobal/HGNRest/pull/718 the team members tasks would not load. 

Screen Shot 2024-01-25 at 7 32 13 PM

However when using the current development backend it would load the team members task but the functionality fails with status code 404

Screen Shot 2024-01-25 at 7 34 26 PM

**Review by KevHnh — DISMISSED** [review]
Hi Luis, thanks for getting back to me with the backend issue. Everything works as intended. Excellent work on updating the UI only after the request is successful.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83934825/1b634371-f86e-405d-ba0d-fb433f7f6a43

**Comment by luisarevalo21** [comment]
> Hi @luisarevalo21, i tested your application and i didn't get to see the team members tasks, look my review please:
> 
>  2024-01-25-13-09-33.mov

Hello 
Please make sure you are checking into the corresponding backend and running `npm run build && npm run start` 
when checking into the branch. then run the frontend with `npm run start:local`
Feel free to reach out via slack for further assistance. 
Thanks

**Comment by luisarevalo21** [comment]
> After testing this PR I came across some problems. When using the backend PR [OneCommunityGlobal/HGNRest#718](https://github.com/OneCommunityGlobal/HGNRest/pull/718) the team members tasks would not load.
> 
> Screen Shot 2024-01-25 at 7 32 13 PM
> However when using the current development backend it would load the team members task but the functionality fails with status code 404
> 
> Screen Shot 2024-01-25 at 7 34 26 PM


Hello 
Please make sure you are checking into the corresponding backend and running `npm run build && npm run start` 
when checking into the branch. then run the frontend with `npm run start:local`
Feel free to reach out via slack for further assistance. 
Thanks

**Comment by luisarevalo21** [comment]
> After testing this PR I came across some problems. When using the backend PR [OneCommunityGlobal/HGNRest#718](https://github.com/OneCommunityGlobal/HGNRest/pull/718) the team members tasks would not load.
> 
> Screen Shot 2024-01-25 at 7 32 13 PM
> However when using the current development backend it would load the team members task but the functionality fails with status code 404
> 
> Screen Shot 2024-01-25 at 7 34 26 PM

Hello 
Please make sure you are checking into the corresponding backend and running `npm run build && npm run start` 
when checking into the branch. then run the frontend with `npm run start:local`
Feel free to reach out via slack for further assistance. 
Thanks

**Review by Sophie-lei — DISMISSED** [review]
Hey! I just tasted this PR. I think the most of parts are working as expected. The warning page pops up and the alert notification is working. However, the Style of the warning button looks different from the video you are sharing. it may need adjusted.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/4b80268e-3c81-4a7b-81cd-81905f1fa50f

**Review by SRamen1999 — DISMISSED** [review]
Hi Luis, thanks for getting back to me. Everything works as intended.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/99db859f-a740-4aa0-8a5b-a24e29065d69

**Comment by xmyoot** [comment]
Hi Luis, awesome work on this PR. I checked out to the two necessary branches and tested #1783 and #718, backend logging correctly but for some reason when I click the same warning multiple times, the state freezes and then I can’t add any more warnings to any users, also I wasn't able to see that modal after the second warning. Please reference the video: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/15240538/fc662ef6-1adb-411b-929f-ae50b1f6bd4d

**Review by 666saofeng — APPROVED** [review]
Hi, I tested PR and  it worked as intended. Congrats!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62363348/0b4929da-b524-4fd3-a004-b805b90a13ac

**Review by BaileyM96 — APPROVED** [review]
Good evening @luisarevalo21 I reviewed your PR and everything looks and works great I am excited to see how this feature will work for everyone.

## Videos


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108101478/80c4ed21-d006-4ad3-a4c9-8cff5da0cda9

**Comment by galgalinidhi** [comment]
Hi @luisarevalo21, I tested the PR and I don't see all the options under Tracking and the Tasks and Timelog component overlaps the Timer.
Screenshot 2024-01-27 at 6 01 20 PM

Screenshot 2024-01-27 at 5 56 04 PM

Also, I see that some of the circles are already selected when I click on Tracking.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44816413/78acd3ca-83c6-4a84-a24b-628e1e7c6934

**Review by peterson337 — DISMISSED** [review]
Hi, I am testing and everything is working as intended.

one comunnity

**Review by X1aoZhang99 — DISMISSED** [review]
Hi Luis,

I test you PR with FE #1873, and I logged in as admin. All functionalities worked as intended, nice work! Please refer to the video. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113070229/b9a3c73e-452b-4329-bbec-6b78a5e9cede

**Comment by luisarevalo21** [comment]
> Hi Luis, awesome work on this PR. I checked out to the two necessary branches and tested #1783 and #718, backend logging correctly but for some reason when I click the same warning multiple times, the state freezes and then I can’t add any more warnings to any users, also I wasn't able to see that modal after the second warning. Please reference the video:
> 
>  PR.1783+.718.mov

Hello 
Please make sure you are checking into the correct backend.
You linked a different Backend hence the errors.

here is the correct backend 
https://github.com/OneCommunityGlobal/HGNRest/pull/718
Once checked in, run the following comand `npm run build && npm run start`
Feel to reach out on slack if you have any further questions.

**Comment by luisarevalo21** [comment]
> Hi @luisarevalo21, I tested the PR and I don't see all the options under Tracking and the Tasks and Timelog component overlaps the Timer. Screenshot 2024-01-27 at 6 01 20 PM
> 
> Screenshot 2024-01-27 at 5 56 04 PM
> Also, I see that some of the circles are already selected when I click on Tracking.
> 
>  Screen.Recording.2024-01-27.at.5.58.13.PM.mov

Hello
It seems you've checked into a previous version of my branch. 
Please fetch from this PR then run ` git pull` then ` npm run install` 

Feel to reach out on slack if you have any further questions.

**Review by pshereen — DISMISSED** [review]
Tested the PR and all the newly implemented features are working as expected.
Logged in as admin user.

Click the toggle button to view the users' warnings - Working
Click an icon to assign a warning to the user' corresponding to the description. - Working
Once they receive more than 2 a modal will appear. - Working
log warning only will color the icon blue like normal - Working
issue warning will color the icon yellow an alert will appear saying an email was sent (Future PR) - Working
Issuing a blue square will color red and an alert will appear saying blue square was sent (Future PR) - Working
Selecting a colored allows you to delete it. - Working
You may toggle the button to remove the warnings component and will appear one you click it again. - Working

Video below shows the working of the features.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/7041610/95c9bd31-07e0-46e2-b5af-a7aaecc5d407

**Review by Pandani07 — DISMISSED** [review]
Tested the PR and implemented features are working as expected.
Logged in as admin user

View warnings by clicking the toggle button - works
Assign respective warning to the user by clicking none colored circle - works
Delete warning by clicking colored circle - works
Modal appears when more than 2 warnings have been given
Scroll-bar in the warnings component works
Warnings hidden after toggle button is clicked - works

Clicking on any circle will not make that particular circle colored, it will increase the number of warnings issues, works as expected

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/c4404fb6-973d-42c7-8810-8fa77fd58c63



Additional web responsiveness checked

**Review by Nahiyan-16 — DISMISSED** [review]
## Great work!
Hi Luis, I tested your PR and I have to say that you did a wonderful job. I logged in as an admin and was able to see the tracking toggle button underneath all users. I was able to access and issue warnings to each row. After issuing two warnings, a pop-up box appears with the correct options and each option works fine. Deleting warnings also works fine. I do have a suggestion for this pop-up box: 
Screenshot 2024-02-01 132122
As you can see there is a text where you are telling the user what the color of the circle will be but it doesn't change at all.  An option could be to update the word or even use a visual object to notify the user upon hovering over an option. 

Otherwise, this feature works as intended:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/c286e7a4-d7f9-4dbd-9f13-b8b83d4504d2

**Review by ClemarNunes — DISMISSED** [review]
I tested your PR and logged in as an administrator. All features worked as intended.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/9a83690d-33a7-4dde-950f-3cbc218c9071

**Review by yvonnep165 — DISMISSED** [review]
Hi @luisarevalo21 ,
I have tested the PR and the implemented features are working as expected. Nice work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/103622026/a1ed9380-d2e3-48e2-9cc0-f8673c6ff1c7

**Review by sucheta90 — DISMISSED** [review]
Hello @luisarevalo21 , I tested this PR as an owner, and was able to issue warnings to another profile. It works as intended. Great work! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/e10a1aee-3f8a-4984-a340-79f7b1df9f4f

**Comment by mohabbasd** [comment]
Hello,
So I had a few issues first and after reading the comment I was able to get the PR review done exactly as intended.
This is the video in which this PR was executed successfully.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91215370/50291908-95f3-492b-9807-b2d69b63e8ac

**Review by mohabbasd — DISMISSED** [review]
Hello,
So I had a few issues first and after reading the comment I was able to get the PR review done exactly as intended.
This is the video in which this PR was executed successfully.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91215370/2c7d8f57-d1e1-4757-aba3-94e62f258939

File: src/components/Warnings/WarningItem.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+function WarningItem({
+  warningText,
+  handlePostWarningDetails,
+  warnings,
+  username,
+  handleDeleteWarning,
+  submitWarning,
+}) {
+  const [toggleModal, setToggleModal] = useState(false);
+  const [warning, setWarning] = useState(null);
```
Comment (Coops023) [inline src/components/Warnings/WarningItem.jsx:17]: Perhaps grouping these in a state object could be an improvement as they all deal with the warning state?
`const [warningState, setWarningState] = useState({
  warning: null,
  deleteWarning: false,
  warningId: null,
});`

**Review by Coops023 — COMMENTED** [review]
When logged as an admin user, i was able to toggle between being able to showing and not showing warnings, add and remove warnings, log or issue warnings and blue squares. All functionality tests went well, great job implementing this feature. 
Screenshot 2024-02-13 at 19 45 36

**Review by XiaohanMeng — COMMENTED** [review]
The admin account testing confirms the application works as intended with the following functionalities:

- Ability to toggle visibility of warnings.
- Assigning warnings is functional.
- Log warnings are correctly displayed in blue.
- Issuing a warning colors the icon yellow, and an alert appears indicating an email was sent.
- Issuing a blue square turns the icon red, and an alert appears indicating a blue square was sent.
- Warnings can be deleted, and the sequence of warnings remains correct even if they are not deleted in order.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/b62b12b1-3be4-4f3b-86ae-c85e382b0f05

**Review by Coops023 — APPROVED** [review]

**Comment by Jingyii800** [comment]
Hi @luisarevalo21, good job!  I tested the warning functions, it works well to 

- Open the "track" toggle button to see the warnings for each volunteer
- Delete the warning 
- Add warnings
- When there are more than 2 warnings, a window will pop up to choose the detailed warning issues
- Hide warnings by clicking on the "hide" button

I only encountered one issue. When I tried to delete one warning but accidentally clicked twice by mistake, there was a warning showing "error occurred when posting". Then I **MUST REFRESH** the whole page to continue working. Please consider adding a feature to partly refresh one volunteer's warning issues instead of refreshing the whole page manually.

Please refer to more details in my video.  

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/f150ed26-c505-4a44-a712-39aea7d1bed4

**Review by Jingyii800 — DISMISSED** [review]

**Review by XiaohanMeng — DISMISSED** [review]

**Comment by one-community** [comment]
Thank you all, moving to final review!

**Review by cvtqx — DISMISSED** [review]
Tested this PR and everything works as described, warnings can be issued and deleted and the changes are reflected in the UI accordingly. Excellent job! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/18cc7e4e-200d-4ace-a676-6100a42aeb51

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**